iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -1395,6 +1397,33 @@ internal static string PopulateMessagePropertiesFromMessage(string topicName, Me
             return msg;
         }
 
+        private static IReadOnlyDictionary<TKey, TValue> MergeDictionaries<TKey, TValue>(IDictionary<TKey, TValue>[] dictionaries)
+        {
+            // No item in the array should be null.
+            if (dictionaries == null || dictionaries.Any(item => item == null))
+            {
+                throw new ArgumentNullException(nameof(dictionaries), "Provided dictionaries should not be null");
+            }
```

Comment:
I think there is an option for this in `Arguments.RequireNotNullOrEmpty<T>()` that I added to the device's Arguments class you could copy to the service one. The source for this was Azure.Core.

Reply:
This class will disappear once we move to MQTTNet